### PR TITLE
[4.0] Fix showOn for select field with multiple attr

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/35486 .

### Summary of Changes
- Fixes the bug
- Removes the IIFE as the script is a module now

### Testing Instructions

- Check the banners config.xml: www/administrator/components/com_banners/config.xml
- The config has the fields save_history (radio) and history_limit (number) showon="save_history:1"
- When you go to the config form, you can enable "Enable Versions" and it shows the "Maximum Versions" associated field
- Change the save_history (radio) to save_history (list), type="list", it still works OK
- Remove the layout to save_history (list), layout="joomla.form.field.radio.switcher", it still works OK
- Add the multiple="true" to the save_history (list), the shown conditional field doesn't work.

Or check https://github.com/joomla/joomla-cms/issues/35486#issuecomment-932502990

### Actual result BEFORE applying this Pull Request

Not working

### Expected result AFTER applying this Pull Request

Working

### Documentation Changes Required

No, bug fix

### CODE REVIEW

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 882434a731ea82f841dbacc225c64b31a93e3dd9<br /><br />Test OK!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on b18a1fe0c840a2e1db1c9e62161445dbd7446517<br /><br />It still works OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by andreagentil** [comment]
I have tested this item :white_check_mark: successfully on b18a1fe0c840a2e1db1c9e62161445dbd7446517<br /><br />It works. :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by bembelimen** [comment]
I assume, that everyone also tested, that the old behaviour (single field) does not break?

@dgrammatiko if you fix the conflicts, I can merge.

**Comment by dgrammatiko** [comment]
@bembelimen  done
@anibalsanchez could you check that the nonmultiple elements work fine as well?

**Comment by anibalsanchez** [comment]
@dgrammatiko I've been working with the patch and it has worked Ok.

File: build/media_source/system/js/showon.es6.js [hunk 1]
```diff
@@ -2,282 +2,280 @@
  * @copyright  (C) 2018 Open Source Matters, Inc. <https://www.joomla.org>
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
-((document) => {
-  'use strict';
```
Comment (Fedik) [inline build/media_source/system/js/showon.es6.js:6]: here also, it not a module ;) 
please keep it

Comment (dgrammatiko) [reply build/media_source/system/js/showon.es6.js:6]: Grrr, the whole system JSON is wrong

**Comment by Fedik** [comment]
still need some changes<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

File: build/media_source/system/js/showon.es6.js (hunk 1 above)
Comment (Fedik) [reply build/media_source/system/js/showon.es6.js:6]: you no need to rewrite everything, you can keep it as it was,
and we can updated it later, when decided to drop whole es5 thing

Comment (dgrammatiko) [reply build/media_source/system/js/showon.es6.js:6]: Ok, done

**Comment by dgrammatiko** [comment]
@anibalsanchez @andreagentil could you revalidate your tests here

File: build/media_source/system/js/showon.es6.js (hunk 1 above)
Comment (dgrammatiko) [reply build/media_source/system/js/showon.es6.js:6]: >  you can keep it as it was,

Actually, we're serving always modern code without the `type=module`. In theory, we break older browsers, and I'm saying in theory as we all know that for these browsers the output is already broken even if ES5 was loaded (missing polyfills, etc). Anyways I'm willing to put some work on it but it's better to have an answer on whether ES5 will still be around or not...

Comment (Fedik) [reply build/media_source/system/js/showon.es6.js:6]: > Actually, we're serving always modern code 

It does not mater how modern the code,
It more for own protection from bad hands or mistake, if someone accidentally define variable or function that will ended up in global scope. 
```
fooBar = 'foo';
function joomla(){}
```
will not show an error, and will add it to `window.fooBar` and `window.joomla`.

For non "module" code `'use strict';` is required 
[Also it enable little optimization.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode)

Comment (dgrammatiko) [reply build/media_source/system/js/showon.es6.js:6]: Sure but type=module is by default strict so no need for `'use strict';` and Rollup makes it IIFE with `'use strict';` for us
- type=module
Screenshot 2021-10-07 at 21 12 36

- nomodule
Screenshot 2021-10-07 at 21 12 50

Comment (dgrammatiko) [reply build/media_source/system/js/showon.es6.js:6]: Also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#other_differences_between_modules_and_standard_scripts

Comment (Fedik) [reply build/media_source/system/js/showon.es6.js:6]: yes, that why I said for "non module" ;)

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on c3c22b6e475054ee0513a3d9a41f52faf7e1975f<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by andreagentil** [comment]
It works.

**Comment by andreagentil** [comment]
I have tested this item :white_check_mark: successfully on c3c22b6e475054ee0513a3d9a41f52faf7e1975f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35723">issues.joomla.org/tracker/joomla-cms/35723</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**